src/libmongoc/tests/unified/entity-map.c

```diff
@@ -1761,6 +2003,28 @@ entity_map_add_bson (entity_map_t *em,
    return _entity_map_add (em, id, "bson", (void *) bson_val_copy (val), error);
 }
 
+bool
+entity_map_add_bson_array (entity_map_t *em,
+                           const char *id,
+                           bson_error_t *error)
+{
+   // Note: the specification states we should be storing a BSON object of array
+   // type, but we use an array of BSON objects instead to make append and
+   // iteration easier.
+   mongoc_array_t *array = BSON_ALIGNED_ALLOC (mongoc_array_t);
```

Comment:
This was a leftover from prior iterations that had attempted to use `bson_t` objects as elements instead of `bson_t*`. Alignment is not necessary for `mongoc_array_t`. Will simplify.